pkg/controller/virtualmachineinstance/controller.go

```diff
@@ -52,7 +65,24 @@ func (h *Handler) OnVmiChanged(_ string, vmi *kubevirtv1.VirtualMachineInstance)
 		return vmi, nil
 	}
 
-	node, err := h.nodeCache.Get(vmi.Name)
+	nodeName := vmi.Name
+	guestAgentInfo, err := h.kubevirtClient.VirtualMachineInstance(vmi.Namespace).GuestOsInfo(context.TODO(), vmi.Name)
+	if err != nil {
+		logrus.WithFields(logrus.Fields{
+			"name":      vmi.Name,
+			"namespace": vmi.Namespace,
+		}).WithError(err).Error("failed to get guest agent info, fallback to use vmi name as node name")
+	} else {
+		logrus.WithFields(logrus.Fields{
+			"name":      vmi.Name,
+			"namespace": vmi.Namespace,
+			"hostname":  guestAgentInfo.Hostname,
+		}).Info("get agent info success, using hostname as node name")
+		nodeName = guestAgentInfo.Hostname
+		h.nodeToVMName.Store(nodeName, vmi.Name)
```

Comment:
The `vmi.Namespace` is not stored in the map, if VMs from different namespace have same name, will them overwrite each other?

Reply:
I think the better way is to drop events from other namespaces because the cloud provider is running in the guest cluster. It only needs to care VMI in that namespace. We can get the namespace like instance manager. WDYT?

https://github.com/harvester/cloud-provider-harvester/blob/9daecd12ec639803a8b27eeb30daec6834cb45cb/pkg/cloud-controller-manager/ccm.go#L85-L89

Reply:
It is reasonable to drop other namespaces.